### ISSUE TITLE
Sprites Not Showing

### ISSUE THREAD

**Issue by xycarto** [openlayers/ol-mapbox-style#331]
Using: ol-mapbox-style@6.3.2

Calling styleJSON with:

```
fetch('./styleText.json').then(function(response) {
  response.json().then(function(glStyle) {
    olms.applyStyle(vectorMap, glStyle, 'placeNames');
  });
});
``` 

With a styleText.json:

```
{
  "version": 8,
  "name": "placeNames",
  "glyphs": "https://orangemug.github.io/font-glyphs/glyphs/{fontstack}/{range}.pbf",
  "sprites": "https://localhost:8000/sprites",
  "sources": {
    "placeNames": {
      "type": "vector"
    }
  },
  "layers": [
    {
        "id": "harbour",
        "type": "symbol",
        "source": "placeNames",
        "source-layer": "placenames_poc.placenames_poc",
        "filter": ["==", "feat_type", "harbour"],
        "layout": {
          "text-field": ["get", "name"],
          "text-font": ["Myriad Pro", "sans-serif Italic"]
        },
        "paint": {
          "text-color": "#5990A8"
        }
    },
    {
      "id": "island",
      "type": "symbol",
      "source": "placeNames",
      "source-layer": "placenames_poc.placenames_poc",
      "filter": ["==", "feat_type", "island"],
      "layout": {
        "icon-image": "accommodation_camping"
      }
    }
  ]
}
```

Sprites are definitely accessible at: https://localhost:8000/sprites and I am certain the Vector Tiles are loading.

I cannot seem to get the sprite icon to show up. I don't get any errors and I've tried any number fo combinations of ` applyStyle ` or ` stylefunction ` . I am certain the styleText JSON is functioning, since the ` harbour ` labels do show and if I try to display labels in the ` "id": "island" ` , these work perfectly well. Declutter on the vector tile is turned off also. 

Maybe I am missing something or could this be the normal functionality of applyStyle?

**Comment by ahocevar** [comment]
Does your localhost server send CORS headers with the response? Lack of doing that is the most common issue of sprites not showing.

**Comment by xycarto** [comment]
Thank you for getting back to me. I tested running the localhost via the npm:

`http-server -p 8000 --cors`

No luck.

So I moved the sprites to a different sever on localhost and served the sprites separately using:

 `http-server -p 3000 --cors`

No luck

Finally the moved the sprites to an external server. 

`https://xycarto.github.io/vectortile-repo/sprites/`

No luck there

I also set up my Firefox to allow cors anywhere, and also tested in Chrome and Safari.

It is almost as if the sprites address is not read in the JSON document or the error is hidden.  Even if I use a trash url in the `style.json` like:

`"sprites": "https://www.foo-bar-trash/"`

I still do not get an error
